### PR TITLE
PVS-proxy updates for performance and error handling

### PR DESCRIPTION
**Pull request by robhoes**

### CODE REVIEW

**Comment by robhoes** [comment]
Depends on https://github.com/xenserver/xcp-idl/pull/9.

File: ocaml/xapi/pvs_proxy_control.ml [hunk 1]
```diff
@@ -19,8 +19,9 @@ open Threadext
 module D = Debug.Make(struct let name = "xapi_pvs_proxy_control" end)
 open D
 
-let proxy_port_name bridge =
-  "pvs-" ^ bridge
+let proxy_port_name vif =
+  let uuid = String.sub vif.API.vIF_uuid 0 8 in
```
Comment (jjd27) [inline ocaml/xapi/pvs_proxy_control.ml:23]: Why only the first 8 characters? Is this due to interface name length restrictions?

**Review by jjd27 — APPROVED** [review]

File: ocaml/xapi/pvs_proxy_control.ml (hunk 1 above)
Comment (robhoes) [reply ocaml/xapi/pvs_proxy_control.ml:23]: Yes indeed, I tried to make it longer, but the kernel didn't accept it.

Comment (jjd27) [reply ocaml/xapi/pvs_proxy_control.ml:23]: Okay. Well, a clash is still pretty unlikely, and it's not a big problem if there is a clash.

**Pull request merged**